samcli/lib/telemetry/cicd.py

```diff
@@ -46,9 +46,19 @@ def _is_codeship(environ: Mapping) -> bool:
     return ci_name == "codeship"
 
 
+def _is_jenkins(environ: Mapping) -> bool:
+    """
+    Use environ to determine whether it is running in Jenkins.
+    According to the doc,
+    https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#working-with-your-jenkinsfile
+    > BUILD_TAG
+    >   String of jenkins-${JOB_NAME}-${BUILD_NUMBER}.
+    """
+    return bool(environ.get("BUILD_TAG", "").startswith("jenkins-"))
+
+
 _ENV_VAR_OR_CALLABLE_BY_PLATFORM: Dict[CICDPlatform, Union[str, Callable[[Mapping], bool]]] = {
-    # https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#using-environment-variables
-    CICDPlatform.Jenkins: "JENKINS_URL",
+    CICDPlatform.Jenkins: _is_jenkins,
```

Comment:
should we do an or condition here? or is build_tag always true and present with the jenkins prefix?

Reply:
According to the doc, it does not have a condition. 
I tested it in my own jenkins instance and can see the variable value did starts with `jenkins-`

Reply:
ok, but should we look for JENKINS_URL or BUILD_TAG with jenkins-, is there a condition under which either one may be present or not, we would want to include both.

Reply:
`BUILD_TAG` itself should suffice, but I don't mind adding a `JENKINS_URL` since it works today already, having it is safer

Reply:
added